### PR TITLE
DOCS-1003: Update `obstacles_pointcloud` attributes

### PR DESCRIPTION
**Pull request by sguequierre**
* Make most attributes optimal, change defaults in table 
* Add new filtering parameter `clustering_strictness`

### CODE REVIEW

File: docs/services/vision/segmentation.md [hunk 1]
```diff
@@ -116,13 +119,14 @@ The following parameters are available for a `"obstacles_pointcloud"`.
...
+| `min_points_in_plane` | Optional | An integer that specifies how many points there must be in a flat surface for it to count as a plane. This is to distinguish between large planes, like the floors and walls, and small planes, like the tops of bottle caps. <br> Default: `500` </br> |
+| `min_points_in_segment` | Optional | An integer that sets a minimum size to the returned objects, and filters out all other found objects below that size. |
+| `clustering_radius` | Optional | An integer that specifies how far apart points can be (in units of mm) in order to be considered part of the same object. A small clustering radius will more likely split different parts of a large object into distinct objects. A large clustering radius may aggregate closely spaced objects into one object. 3.0 is a decent starting value. <br> Default: `1` </br> |
```
Comment (bhaney) [inline docs/services/vision/segmentation.md:124]: The radius is no longer in units of mm, as the underlying algorithm has fundamentally changed. The description should be updated to just read:
"An integer that specifies which neighboring points count as being "close enough" to be potentially put in the same cluster. This parameter determines how big candidate clusters should be. A small clustering radius will more likely split different parts of a large object into distinct objects. A large clustering radius may aggregate closely spaced objects into one object."

**Review by bhaney — CHANGES_REQUESTED** [review]
Thanks! Just a few  changes

**Comment by bhaney** [comment]
@kharijarrett can approve this PR while I am out of office

**Review by bhaney — COMMENTED** [review]
commenting so my "request changes" does not block merging

**Review by bhaney — COMMENTED** [review]
commenting so my "request changes" does not block merging

**Review by bhaney — APPROVED** [review]

File: docs/services/vision/segmentation.md (hunk 1 above)
Comment (sguequierre) [inline docs/services/vision/segmentation.md:124]:
```suggestion
| `clustering_radius` | Optional | An integer that specifies which neighboring points count as being "close enough" to be potentially put in the same cluster. 
This parameter determines how big the candidate clusters should be. 
A small clustering radius is likely to split different parts of a large cluster into distinct pointcloud objects. 
A large clustering radius is likely to aggregate closely spaced clusters into one pointcloud object. <br> Default: `1` </br> |
```

File: docs/services/vision/segmentation.md [hunk 2]
```diff
@@ -116,13 +119,17 @@ The following parameters are available for a `"obstacles_pointcloud"`.
...
+| `clustering_radius` | Optional | An integer that specifies which neighboring points count as being "close enough" to be potentially put in the same cluster. 
+This parameter determines how big the candidate clusters should be. 
+A small clustering radius is likely to split different parts of a large cluster into distinct pointcloud objects. 
+A large clustering radius is likely to aggregate closely spaced clusters into one pointcloud object. <br> Default: `1` </br> |
+| `clustering_strictness` | Optional | An integer that determines the probability threshold for sorting neighboring points into the same cluster, or how "easy" `viam-server` should determine it is to sort the points the robot's camera sees into this pointcloud.
+The lower the number, the bigger your clusters will be. <br> Default: `5` </br> |
+| `max_dist_from_plane_mm` | Optional | A float that determines how much area above and below an ideal ground plane should count as the plane for which points are removed. For fields with tall grass, this should be a high number. The default value is 100 mm. <br> Default: `1`. </br> |
```
Comment (sguequierre) [inline docs/services/vision/segmentation.md:130]:
```suggestion
| `max_dist_from_plane_mm` | Optional | A float that determines how much area above and below an ideal ground plane should count as the plane for which points are removed. For fields with tall grass, this should be a high number. The default value is 100 mm. <br> Default: `100`. </br> |
```

File: docs/services/vision/segmentation.md [hunk 3]
```diff
@@ -116,13 +119,14 @@ The following parameters are available for a `"obstacles_pointcloud"`.
...
+| `min_points_in_segment` | Optional | An integer that sets a minimum size to the returned objects, and filters out all other found objects below that size. |
+| `clustering_radius` | Optional | An integer that specifies how far apart points can be (in units of mm) in order to be considered part of the same object. A small clustering radius will more likely split different parts of a large object into distinct objects. A large clustering radius may aggregate closely spaced objects into one object. 3.0 is a decent starting value. <br> Default: `1` </br> |
+| `clustering_strictness` | Optional | An integer that determines the probability threshold for sorting neighboring points into the same cluster, or how "easy" `viam-server` should determine it is to sort the points the robot's camera sees into this pointcloud.
+The lower the number, the bigger your clusters will be. <br> Default: `5` </br> |
+| `max_dist_from_plane_mm` | Optional | A float that determines how much area above and below an ideal ground plane should count as the plane for which points are removed. For fields with tall grass, this should be a high number. The default value is 100 mm. <br> Default: `1`. </br> |
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/services/vision/segmentation.md:130]: Changed, might have been a copy error-- thank you for catching!

File: docs/services/vision/segmentation.md [hunk 4]
```diff
@@ -116,13 +119,17 @@ The following parameters are available for a `"obstacles_pointcloud"`.
...
-| `ground_angle_tolerance_degs` | Optional | A float that determines how strictly the found ground plane should match the `ground_plane_normal_vec`. For example, even if the ideal ground plane is purely flat, a rover may encounter slopes and hills. The algorithm should find a ground plane even if the found plane is at a slant, up to a certain point. <br> Default: `30.00` </br> |
+| `min_points_in_plane` | Optional | An integer that specifies how many points there must be in a flat surface for it to count as a plane. This is to distinguish between large planes, like the floors and walls, and small planes, like the tops of bottle caps. <br> Default: `500` </br> |
+| `min_points_in_segment` | Optional | An integer that sets a minimum size to the returned objects, and filters out all other found objects below that size. |
+| `clustering_radius` | Optional | An integer that specifies which neighboring points count as being "close enough" to be potentially put in the same cluster.
```
Comment (sguequierre) [inline docs/services/vision/segmentation.md:124]: @kharijarrett / Bijan LMK what you think of my wording change w pointcloud here, I'm not sure about technical accuracy

Comment (kharijarrett) [reply docs/services/vision/segmentation.md:124]: Mildly annoying but can it say "how many points there must be **on** a flat surface..."

Comment (sguequierre) [reply docs/services/vision/segmentation.md]: and don't worry 😆 That's a great suggestion

File: docs/services/vision/segmentation.md [hunk 5]
```diff
@@ -121,10 +121,7 @@ The following parameters are available for a `"obstacles_pointcloud"`.
 | --------- | --------- | ----------- |
 | `min_points_in_plane` | Optional | An integer that specifies how many points there must be in a flat surface for it to count as a plane. This is to distinguish between large planes, like the floors and walls, and small planes, like the tops of bottle caps. <br> Default: `500` </br> |
```
Comment (kharijarrett) [inline docs/services/vision/segmentation.md:122]: I meant it here in "min_points_in_plane" don't hate me lol

**Review by kharijarrett — APPROVED** [review]
LGTM!

File: docs/services/vision/segmentation.md (hunk 5 above)
Comment (sguequierre) [reply docs/services/vision/segmentation.md]: @kharijarrett Hated (/s) 🤣 I updated but lmk what you think of the update to `min_points_in_plane`, I'm still not sure if I'm phrasing it well!

**Pull request merged**